### PR TITLE
Add events for proposal codex

### LINKED ISSUE #2126: Codex events for proposal creation
We need this for the query node, a again with full payloads possible, the events in the engine are opaque.

### PR DESCRIPTION
**Pull request by conectado**
# Fixes #2126 

This PR:
* Adds events for proposal creation in proposal codex
* Adds events for the other extrinsics in proposal codex
* Tests for the events in benchmarks and tests
* Add tests for the execution extrinsics(Note: there is no test for runtime upgrade succeeding)

### CODE REVIEW

File: runtime-modules/proposals/codex/src/lib.rs [hunk 1]
```diff
@@ -255,6 +258,21 @@ pub type GeneralProposalParameters<T> = GeneralProposalParams<
...
 
+decl_event! {
+    pub enum Event<T> where
+        GeneralProposalParameters = GeneralProposalParameters<T>,
+        ProposalDetailsOf = ProposalDetailsOf<T>,
+        WorkingGroup = WorkingGroup,
+        Balance = BalanceOf<T>,
+        BalanceKind = BalanceKind
+    {
+        ProposalCreated(GeneralProposalParameters, ProposalDetailsOf),
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/codex/src/lib.rs:273]: Comment events, please.

File: runtime-modules/proposals/codex/src/lib.rs [hunk 2]
```diff
@@ -437,7 +457,11 @@ decl_module! {
             Self::ensure_details_checks(&proposal_details)?;
 
             let proposal_parameters = Self::get_proposal_parameters(&proposal_details);
-            let proposal_code = T::ProposalEncoder::encode_proposal(proposal_details);
+            // TODO: encode_proposal could take a reference instead of moving to prevent cloning
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/codex/src/lib.rs:479]: Makes sense. Create an issue, please, and add it to the comment.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks good. Left a couple of minor comments.

File: runtime-modules/proposals/codex/src/lib.rs (hunk 2 above)
Comment (conectado) [reply runtime-modules/proposals/codex/src/lib.rs:479]: Done #2161

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Pull request merged**